pkg/db/db.go

```diff
@@ -310,71 +310,84 @@ func (h *DBHandler) DBReadEslEventInternal(ctx context.Context, tx *sql.Tx, firs
 		selectQuery,
 	)
 	if err != nil {
-		return nil, fmt.Errorf("could not query esl table from DB. Error: %w\n", err)
+		return nil, fmt.Errorf("could not query event_sourcing_light table from DB. Error: %w\n", err)
 	}
 	defer func(rows *sql.Rows) {
 		err := rows.Close()
 		if err != nil {
 			logger.FromContext(ctx).Sugar().Warnf("row closing error: %v", err)
 		}
 	}(rows)
+	var row = &EslEventRow{
+		EslId:     0,
+		Created:   time.Unix(0, 0),
+		EventType: "",
+		EventJson: "",
+	}
 	if rows.Next() {
-		var row = EslEventRow{
-			EslId:     0,
-			Created:   time.Unix(0, 0),
-			EventType: "",
-			EventJson: "",
-		}
 		err := rows.Scan(&row.EslId, &row.Created, &row.EventType, &row.EventJson)
 		if err != nil {
 			if errors.Is(err, sql.ErrNoRows) {
 				return nil, nil
 			}
 			return nil, fmt.Errorf("Error scanning event_sourcing_light row from DB. Error: %w\n", err)
 		}
-		logger.FromContext(ctx).Sugar().Warnf("read row: %v", row)
-		return &row, nil
 	}
-	return nil, nil // no rows, but also no error
+	err = rows.Close()
+	if err != nil {
```

Comment:
Isn't this rows.close (and dealing with the error) treated on the defer function?

Reply:
Yes, but in a `defer`, we don't have a good way to return the error to the caller.
Therefore we use `defer` here only as a "catch all", and the proper error handling happens outside of `defer`.